src/genie/libs/parser/asa/show_route.py

```diff
@@ -326,3 +326,156 @@ def cli(self, output=None):
                 index += 1
 
         return ret_dict
+
+
+# =============================================
+# Parser for 'show route bgp'
+# =============================================
+class ShowRouteBgp(ShowRouteSchema):
+    """Parser for
+        * show route bgp
+    """
+
+    cli_command = 'show route bgp'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        ret_dict = {}
+        # strip this patter from the original text
+        # Codes: L - Local, C - connected, S - static, I - IGRP, R - RIP, M - mobile, B - BGP
+        # D - EIGRP, E - EGP, EX - EIGRP external, O - OSPF, I - IGRP, IA - OSPF inter area
+        # N1 - OSPF NSSA external type 1, N2 - OSPF NSSA external type 2
+        # E1 - OSPF external type 1, E2 - OSPF external type 2, E - EGP
+        # i - IS-IS, L1 - IS-IS level-1, L2 - IS-IS level-2, ia - IS-IS inter area
+        # * - candidate default, su - IS-IS summary, U - per-user static route, o - ODR
+        # P - periodic downloaded static route, + - replicated route
+        # SI - Static InterVRF
+        p1 = re.sub(r'(?ms)(Codes:.+?)replicated\sroute(\s*SI\s-\sStatic\sInterVRF)?', '', out)
+
+        lines = [x.strip() for x in p1.splitlines()]
+        entries = dict()
+        last_entry = str()
+        clean_lines = list()
+
+        for line in lines:
+            if re.match(r'(^[A-Z]{1,2})', line):
+                entries[line] = list()
+                last_entry = line
+            else:
+                if last_entry == '':
+                    continue
+
+                entries[last_entry].append(line)
+
+        for k, v in entries.items():
+            clean_lines.append(' '.join(k.split()) + " " + " ".join(v))
+
+        # clean_lines now holds a list of entries in a single line (easier for parsing)
```

Comment:
While I don't necessarily disagree with cleaning up the output before parsing, I can't help but feel this makes it more difficult of a process for both yourself and anyone else who might need to make changes to this parser in the future. It also introduces extra failure points should the output change in any way